### PR TITLE
Changes for #3028

### PR DESCRIPTION
**Pull request by davidsoncasey**
I've overwritten the `__contains__` method of `RequestsCookieJar` as discussed in the issue. It looks to be working as expected and tests pass, but this is my first contribution to this project so I could be missing something.

### CODE REVIEW

File: tests/test_requests.py [hunk 1]
```diff
@@ -754,6 +754,30 @@ def test_cookie_as_dict_items(self):
...
+        value = 'some_value'
+        domain1 = 'test1.com'
+        domain2 = 'test2.com'
+
+        jar = requests.cookies.RequestsCookieJar()
+        jar.set(key, value, domain=domain1)
+        jar.set(key, value, domain=domain2)
+        assert key in jar
+        items = jar.items()
+        assert len(items) == 2
```
Comment (Lukasa) [inline tests/test_requests.py:768]: Can this test also validate that we get a CookieConflictError here too?

File: tests/test_requests.py [hunk 2]
```diff
@@ -754,6 +754,30 @@ def test_cookie_as_dict_items(self):
...
+
+    def test_cookie_duplicte_names_raises_cookie_conflict_error(self):
```
Comment (Lukasa) [inline tests/test_requests.py]: Typo of duplicate.

**Comment by Lukasa** [comment]
Thanks for this @davidsoncasey, this pretty much looks right! I left a few small notes inline for you to take a look at, but one those are addressed we can merge this!

**Comment by davidsoncasey** [comment]
@Lukasa great, thanks for taking a look! I'll have a chance later today to make those edits.

**Comment by davidsoncasey** [comment]
@Lukasa I added a couple fixes, take a look and let me know if there's anything else!

**Comment by Lukasa** [comment]
\o/ Thanks so much @davidsoncasey! :sparkles: :cake: :sparkles:

**Pull request merged**